src/views/Proposal/Votes.vue

```diff
@@ -0,0 +1,218 @@
+
+const network = computed(() => getNetwork(props.proposal.network));
+const votingPowerDecimals = computed(() => {
+  return Math.max(
+    ...props.proposal.space.strategies_parsed_metadata.map(metadata => metadata.decimals),
+    0
+  );
+});
+
+function reset() {
+  votes.value = [];
+  loaded.value = false;
+  loadingMore.value = false;
+  hasMore.value = false;
+}
+
+async function loadVotes() {
+  votes.value = await network.value.api.loadProposalVotes(
+    props.proposal,
+    { limit: LIMIT },
+    choiceFilter.value,
+    sortBy.value
+  );
+  hasMore.value = votes.value.length === LIMIT;
+  loaded.value = true;
+}
+
+function handleSortChange(type: 'vp' | 'created') {
+  if (sortBy.value.startsWith(type)) {
+    sortBy.value = sortBy.value.endsWith('desc') ? `${type}-asc` : `${type}-desc`;
+  } else {
+    sortBy.value = `${type}-desc`;
+  }
+}
+
+async function handleEndReached() {
+  if (loadingMore.value || !hasMore.value) return;
+
+  loadingMore.value = true;
+  const newVotes = await network.value.api.loadProposalVotes(
+    props.proposal,
+    {
+      limit: LIMIT,
+      skip: votes.value.length
+    },
+    choiceFilter.value,
+    sortBy.value
+  );
+  hasMore.value = newVotes.length === LIMIT;
+  votes.value = [...votes.value, ...newVotes];
+  loadingMore.value = false;
+}
+
+onMounted(() => {
+  loadVotes();
+});
+
+watch(
+  () => props.proposal.id,
+  (toId, fromId) => {
+    if (toId === fromId) return;
+
+    reset();
+  }
+);
+
+watch([sortBy, choiceFilter], () => {
+  reset();
+  loadVotes();
+});
+</script>
+
+<template>
+  <div class="flex mx-4 my-4 gap-3 justify-between">
+    <UiSelect
+      v-model="choiceFilter"
+      title="Choice"
+      gap="12px"
+      placement="left"
+      :items="[
+        { key: 'any', label: 'Any' },
+        { key: 'for', label: 'For', indicator: 'bg-choice-for' },
+        { key: 'against', label: 'Against', indicator: 'bg-choice-against' },
+        { key: 'abstain', label: 'Abstain', indicator: 'bg-choice-abstain' }
+      ]"
+    />
+  </div>
+
+  <BlockInfiniteScroller :loading-more="loadingMore" @end-reached="handleEndReached">
+    <table class="text-right w-full table-fixed">
+      <colgroup>
+        <col class="w-[50%] lg:w-[40%]" />
+        <col class="w-[25%] lg:w-[20%]" />
+        <col class="w-[25%] lg:w-[20%]" />
+        <col class="w-[60px] lg:w-[20%]" />
+        <col class="w-[0px] lg:w-[100px]" />
+      </colgroup>
+      <thead>
+        <tr class="border-b">
+          <th class="text-left pl-4 eyebrow text-skin-text pb-2">Votes</th>
+          <th class="eyebrow text-skin-text">
+            <div
+              class="flex justify-end items-center"
+              :class="{
+                'text-skin-link': sortBy === 'created-desc' || sortBy === 'created-asc'
+              }"
+              @click="handleSortChange('created')"
+            >
+              <span>Date</span>
+              <IH-arrow-sm-down v-if="sortBy === 'created-desc'" />
```

Comment:
Should we hide arrow when filter for the column is not selected?

Reply:
I followed the design from [there](https://github.com/snapshot-labs/sx-ui/pull/765#issuecomment-1746845239) which shows icon on multiple comments at the same time (inactive are just grayed out).

Not sure if hiding it might make sense as it won't be clear if that column supports sorting or not (for example Choice is not sortable and doesn't have those arrows).

Reply:
I see, there is a comment there by Fabien though:
> If the sorting is not active we shouldn't show the icon.

Reply:
Oh I see, didn't notice it. Thanks.